Rubberduck.Parsing/ComReflection/ComParameter.cs

```diff
@@ -1,135 +1,53 @@
-using TYPEKIND = System.Runtime.InteropServices.ComTypes.TYPEKIND;
 
 namespace Rubberduck.Parsing.ComReflection
 {
-    [DebuggerDisplay("{DeclarationName}")]
-    public class ComParameter
+    [DebuggerDisplay("{" + nameof(Name) + "}")]
+    public class ComParameter : ComTypedElement
     {
-        public string Name { get; private set; }
+        public override string Name => base.Name ?? $"{Index}unnamedParameter";
+        public override DeclarationType Type => DeclarationType.Parameter;
+        public bool IsByRef { get; }
+        public bool IsOptional { get; }
+        public bool IsReturnValue { get; }
+        public bool IsParamArray { get; set; }
+        public object DefaultValue { get; }
 
-        public string DeclarationName
+        public string DefaultAsEnumMember
         {
             get
             {
-                return string.Format("{0}{1} {2} As {3}{4}{5}",
-                    IsOptional ? "Optional " : string.Empty,
-                    IsByRef ? "ByRef" : "ByVal",
-                    Name,
-                    TypeName,
-                    IsOptional && DefaultValue != null ? " = " : string.Empty,
-                    IsOptional && DefaultValue != null ?
-                        IsEnumMember ? DefaultAsEnum : DefaultValue
-                        : string.Empty);
+                if (!IsEnumMember || !ComProject.KnownEnumerations.TryGetValue(EnumGuid, out ComEnumeration enumType))
+                {
+                    return string.Empty;
+                }
+                var member = enumType.Members.FirstOrDefault(m => m.Value == DefaultValue);
+                return member != null ? member.Name : string.Empty;
             }
         }
 
-        public bool IsArray { get; private set; }
-        public bool IsByRef { get; private set; }
-        public bool IsOptional { get; }
-        public bool IsReturnValue { get; }
-        public bool IsParamArray { get; set; }
-
-        private Guid _enumGuid = Guid.Empty;
-        public bool IsEnumMember => !_enumGuid.Equals(Guid.Empty);
-
-        public object DefaultValue { get; }
-        public string DefaultAsEnum { get; }
-
-        private string _type = "Object";
-        public string TypeName => IsArray ? $"{_type}()" : _type;
-
-        public ComParameter(ELEMDESC elemDesc, ITypeInfo info, string name)
+        public ComParameter(ITypeInfo info, ELEMDESC elemDesc, string name, int index) : base(info, elemDesc, name, index)
         {
-            Debug.Assert(name != null, "Parameter name is null");
-
-            Name = name;
             var paramDesc = elemDesc.desc.paramdesc;
-            GetParameterType(elemDesc.tdesc, info);
             IsOptional = paramDesc.wParamFlags.HasFlag(PARAMFLAG.PARAMFLAG_FOPT);
             IsReturnValue = paramDesc.wParamFlags.HasFlag(PARAMFLAG.PARAMFLAG_FRETVAL);
+            IsByRef = (VarEnum) elemDesc.tdesc.vt == VarEnum.VT_PTR;
```

Comment:
Isn't `VarEnum` used like a bitmask and thus needs to be `And'`d here?

Reply:
Yep. That needs to be fixed.